### ISSUE TITLE
Simple math usage is cumbersome

### ISSUE THREAD

**Issue by niemeyer** [dgraph-io/dgraph#1464]
This is the minimum I could get to work for having a math field inside a query:
```
      f as float
      m as math(f + 1)
      v : val(m)
```
Trying to remove any of these makes the server complain, and that ends up with the same result returned twice: one as val(m) and one as v.

Certainly not high-priority, but worth a look at some point.

**Comment by pawanrawal** [comment]
You could also do 
```
      f as float
      m: math(f + 1)
```

That is you can retrieve the math value by either assigning it to a variable (if you plan to use it later) or you can give it an alias to retrieve it.

**Comment by niemeyer** [comment]
That returns `m not allowed multiple times in same sub-query.`.

As I said, any attempt to make this simpler seems to fail with a different error message.

**Comment by manishrjain** [comment]
Hey @niemeyer ,

Looks like you might be using `m` somewhere else. Can you print the whole query? Also, `dgraph --version`.

**Comment by pawanrawal** [comment]
A sample query that works with `v0.8.1` can be seen at https://play.dgraph.io/0x47b764.

**Comment by niemeyer** [comment]
Apparently the bug is only triggered if there are more fields after the math entry.

Here is your example with an extra query for _name_ after the math: https://play.dgraph.io/0x47b768

It then fails on the math node with: _score not allowed multiple times in same sub-query_

**Comment by pawanrawal** [comment]
Thanks for sharing the query @niemeyer. I have pushed a fix as part of cb8a79fc7ea51b9043ee5d742df22312a47d2900 and it would be part of the next release.
